test/resources/microshift-host.resource

```diff
@@ -158,3 +158,40 @@ Command Should Fail
     ...    return_stderr=True
     ...    return_stdout=True
     Should Not Be Equal As Integers    0    ${rc}
+
+Get System Architecture
+    [Documentation]    Returns the system architecture
+    ${arch}=    Command Should Work    uname -m
+    RETURN    ${arch}
+
+Get MicroShift Cluster ID From File
+    [Documentation]    Read and return the cluster ID from the file.
+    ${id}=    Command Should Work    cat /var/lib/microshift/cluster-id
+
+    Should Not Be Empty    ${id}
+    RETURN    ${id}
+
+Get Deployment Type
+    [Documentation]    Returns whether the target system is bootc, ostree or rpm
+    ${is_bootc}=    Is System Bootc
+    ${is_ostree}=    Is System Ostree
+    ${cluster_id}=    Get MicroShift Cluster ID From File
+
+    ${deployment_type}=    Set Variable    rpm
+    IF    ${is_bootc} == ${TRUE}
+        ${deployment_type}=    Set Variable    bootc
+    ELSE IF    ${is_ostree} == ${TRUE}
+        ${deployment_type}=    Set Variable    ostree
+    END
+    RETURN    ${deployment_type}
```

Comment:
I think this would work, but I don't want to hold PR on this as it's just my 1 cent

```suggestion
    IF    ${is_bootc} == ${TRUE}
        RETURN    bootc
    ELSE IF    ${is_ostree} == ${TRUE}
        RETURN    ostree
    END
    RETURN    rpm
```

Reply:
Done